sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/MergingActiveWindowSet.java

```diff
@@ -124,7 +126,20 @@ public MergingActiveWindowSet(WindowFn<Object, W> windowFn, StateInternals<?> st
   }
 
   @Override
-  public void removeEphemeralWindows() {
+  public void garbageCollect() {
+    // Any remaining NEW windows must have been for closed windows and can be forgotten (again).
```

Comment:
I feel like this comment is misplaced -- it depends on *when* garbageCollect is called. From the perspective here, all new windows are collected. The "any remaining new windows must be for closed windows" belongs at the point we're calling garbageCollect.